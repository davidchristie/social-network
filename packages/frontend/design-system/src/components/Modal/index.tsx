import Dialog from "@material-ui/core/Dialog";
import React from "react";
import Section from "../Section";

interface Props {
  onClose: () => void;
  open: boolean;
}

export default class Modal extends React.Component<Props> {
  public render () {
    if (!this.props.open) {
      return null;
    }
    return (
        <Dialog
          className="Modal"
          onClose={this.props.onClose}
          open={true}
        >
          <Section>
          {this.props.children}
          </Section>
        </Dialog>
    );
  }
}
