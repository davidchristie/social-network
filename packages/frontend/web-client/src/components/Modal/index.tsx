import Dialog from "@material-ui/core/Dialog";
import { Section } from "design-system";
import React from "react";

interface Props {
  onClose: () => void;
  open: boolean;
}

export default class Modal extends React.Component<Props> {
  public render () {
    if (!this.props.open) {
      return null;
    }
    return (
        <Dialog
          className="Modal"
          onClose={this.props.onClose}
          open={true}
        >
          <Section>
          {this.props.children}
          </Section>
        </Dialog>
    );
  }
}
