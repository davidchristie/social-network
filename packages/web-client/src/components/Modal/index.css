.Modal {
  background: white;
  border: 1px solid black;
  box-shadow: 4px 4px 5px rgba(0, 0, 0, 0.4);
  margin: 15vh auto;
  max-width: 600px;
  padding: 1em;
  width: 80%;
}
