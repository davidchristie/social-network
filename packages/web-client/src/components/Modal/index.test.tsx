import React from "react";

import itMatchesSnapshot from "../../testing/itMatchesSnapshot";
import Modal from "./index";

describe("Modal component", () => {
  describe("when closed", () => {
    itMatchesSnapshot(<Modal onClose={() => { return; }} open={false} />);
  });
});
