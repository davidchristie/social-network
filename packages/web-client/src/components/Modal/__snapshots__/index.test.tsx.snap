// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component when closed matches snapshot 1`] = `""`;
